pkg/miniogw/gateway-storj.go

```diff
@@ -100,33 +101,33 @@ func (s *storjObjects) GetBucketInfo(ctx context.Context, bucket string) (
 	return minio.BucketInfo{Name: bucket, Created: meta.Created}, nil
 }
 
-func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
-	startOffset int64, length int64, writer io.Writer, etag string) (err error) {
-	defer mon.Task()(&ctx)(&err)
+func (s *storjObjects) getObject(ctx context.Context, bucket, object string, startOffset int64,
+	length int64) (r io.ReadCloser, err error) {
 	o, err := s.storj.bs.GetObjectStore(ctx, bucket)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	rr, _, err := o.Get(ctx, paths.New(object))
 	if err != nil {
-		return err
+		return nil, err
 	}
-	defer func() {
-		if err := rr.Close(); err != nil {
-			// ignore for now
-		}
-	}()
-	r, err := rr.Range(ctx, startOffset, length)
+
+	defer utils.LogClose(rr)
```

Comment:
Unfortunately, we cannot do this deferred close of the ranger here. It will be executed when the `getObject` method returns and before we actually read from the `r` reader in the `GetObject` and `CopyObject` methods. This means we will be reading from an already closed connection, which will fail.

The best we can do in this helper `getObject` method is to execute `bs.GetObjectStore` and `o.Get` and return the `rr` ranger. It's tinier, but still makes sense to have it as a helper method. Then all deferred closes should be executed in the `GetObject` and `CopyObject` methods.